### PR TITLE
Osgi take 2

### PR DESCRIPTION
**Pull request by iocanel**

### CODE REVIEW

File: kubernetes-client-with-deps/pom.xml [hunk 1]
```diff
@@ -27,10 +27,23 @@
...
   <name>Fabric8 :: Kubernetes :: Java Client with Dependencies</name>
+  <packaging>bundle</packaging>
 
   <properties>
     <javadoc.include.deps>true</javadoc.include.deps>
     <javadoc.source.includes>io.fabric8:kubernetes-client</javadoc.source.includes>
+
+    <osgi.import>*</osgi.import>
+    <osgi.export>
+      io.fabric8.kubernetes.client*"
```
Comment (jimmidyson) [inline kubernetes-client-with-deps/pom.xml]: What's that trailing "?

Comment (iocanel) [inline kubernetes-client-with-deps/pom.xml]: typo

File: openshift-client-with-deps/pom.xml [hunk 2]
```diff
@@ -41,14 +52,24 @@
...
+          </archive>
+        </configuration>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-shade-plugin</artifactId>
         <version>2.4.1</version>
         <configuration>
-          <!-- put your configurations here -->
+          &lt;!&ndash; put your configurations here &ndash;&gt;
```
Comment (jimmidyson) [inline openshift-client-with-deps/pom.xml]: isn't this a proper XML comment?

File: openshift-client-with-deps/pom.xml [hunk 3]
```diff
@@ -27,10 +27,21 @@
...
   <artifactId>openshift-client-with-deps</artifactId>
   <name>Fabric8 :: Openshift :: Java Client with Dependencies</name>
+  <packaging>bundle</packaging>
 
   <properties>
     <javadoc.include.deps>true</javadoc.include.deps>
     <javadoc.source.includes>io.fabric8:openshift-client</javadoc.source.includes>
+    <osgi.import>*</osgi.import>
+    <osgi.export>
+      io.fabric8.openshift.client*"
```
Comment (jimmidyson) [inline openshift-client-with-deps/pom.xml]: trailing " again

File: openshift-client-with-deps/pom.xml [hunk 4]
```diff
@@ -62,18 +83,17 @@
                 <includes>
                   <include>io.fabric8:kubernetes-client</include>
```
Comment (jimmidyson) [inline openshift-client-with-deps/pom.xml]: I guess this means there is no feature dependency between the two now?

File: platforms/karaf/features/src/main/resources/feature.xml [hunk 5]
```diff
@@ -27,18 +27,15 @@
...
     <bundle dependency='true'>mvn:com.fasterxml.jackson.jaxrs/jackson-jaxrs-json-provider/${jackson.version}</bundle>
     <bundle dependency='true'>mvn:com.fasterxml.jackson.dataformat/jackson-dataformat-yaml/${jackson.version}</bundle>
-    <bundle dependency='true'>mvn:io.netty/netty/${netty.version}</bundle>
-    <bundle dependency='true'>mvn:com.ning/async-http-client/${async.http.version}</bundle>
-    <bundle dependency='true'>mvn:org.apache.servicemix.bundles/org.apache.servicemix.bundles.oauth-provider/${oauth.bundle.version}</bundle>
     <bundle dependency='true'>mvn:org.yaml/snakeyaml/${snakeyaml.version}</bundle>
 
     <bundle>mvn:io.fabric8/kubernetes-model/${kubernetes.model.version}</bundle>
-    <bundle>mvn:io.fabric8/kubernetes-client/${project.version}</bundle>
+    <bundle>mvn:io.fabric8/kubernetes-client/${project.version}/jar/bundle</bundle>
```
Comment (jimmidyson) [inline platforms/karaf/features/src/main/resources/feature.xml]: these should be with-deps variants?

File: platforms/karaf/features/src/main/resources/feature.xml [hunk 6]
```diff
@@ -27,18 +27,15 @@
...
   </feature>
 
   <feature name="openshift-client" description="Fabric8 Openshift Client" version="${project.version}">
     <feature>kubernetes-client</feature>
-    <bundle>mvn:io.fabric8/openshift-client/${project.version}</bundle>
+    <bundle>mvn:io.fabric8/openshift-client/${project.version}/jar/bundle</bundle>
```
Comment (jimmidyson) [inline platforms/karaf/features/src/main/resources/feature.xml]: with-deps variant?

File: openshift-client-with-deps/pom.xml (hunk 4 above)
Comment (iocanel) [inline openshift-client-with-deps/pom.xml]: Just ignore the -with-deps they are eventually removed.

File: platforms/karaf/features/src/main/resources/feature.xml (hunk 6 above)
Comment (jimmidyson) [inline platforms/karaf/features/src/main/resources/feature.xml]: the with-deps modules are removed from parent pom but files still exist?

Comment (iocanel) [inline platforms/karaf/features/src/main/resources/feature.xml]: don't ask me, you removed them :-)

I didn't do anything additional with them.

**Comment by iocanel** [comment]
[merge]

**Pull request merged**